### DISCUSSION TITLE
Viking 721

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#91] [General]
I think I've used this.  Plastic transparencies with visible gridwires about a cm apart.  Ran microplato IIRC using 8in (?) floppies.
![fFZtTOIQ2P_071712.png](https://github.com/kej715/DtCyber/assets/136194499/4a880b33-a4c7-49d4-9a14-a384eb3ba02b)

The equates and memory map are nice.  One of these have a full listing?  Haven't touched z80 since my vector graphics 5005 (z80b, 192k, hard sector 5.25 floppy, 3 terms, etc, s-100, very well built chassis, backplane, and boards, but terminals were wonky as hell, would sparkle when humid, and they used baseband over ribbon cable, 50 foot ribbon cable, which when rolled up, makes a really nice loading coil for the unrolled parts, wiping out the entire apt complex phone lines, live and learn, that was embarrassing. Still, it also had a nice WELL FILTERED linear psu, and they didn't skimp on the iron), but unless CDC relied on known z80 bugs, some did, it is a simple instruction set.  There are emulators up the gazoo in public domain, some include all known z-80 bugs as well, iirc.  IMHO, anything that did / can / will do a z80 emulator with cp/m capability, it will have to emulate the bugs too, even if not necessary for the terminal itself.  Too many people outside of computer companies used the bugs at some point, they were well known.  How many BBSs?

**Comment by KC5VDJ** [comment]
Split baud will be platform dependent.  Is it required for anything discussed previously (should I put it on the front burner or back burner)?  There were modems and such with slow back channels, as low as military tty, 75 baud.  Was 75 baud ever used for anything other than 5-bit military ITA2 ttys or modem backchannel?  Kinda like 150 baud with selectrics.  WU ASCII ttys were 110.

Late edit: What the hell used 200?  600 was used for "fast" stuff, being maximum Bell 103 fsk, often borderline at that based on outside plant.  200 baud is throwing me tho, what the hell used it, and are any in working condition, or condition such that repair could be possible at some future date?  Like I said, if I can find that selectric, and have the cash, I'll need 150 baud myself.  If I have my way, even 20 / 40 mA loop support.  I'm kinda focused on the hobbyist/collector thing.

**Reply by kej715** [reply]
Does an ordinary terminal emulator care about baud rate? The 2-port mux emulation in DtCyber uses a TCP connection and expects terminal emulators to connect to that. Same for DtCyber's 6676 mux emulation, and its NPU and MDI emulations. When a Viking 721 was being used as a console, it was normally operating at 19.2Kbps. A TCP connection should provide much higher transfer rate than that.

**Reply by KC5VDJ** [reply]
Ok, tcp won't have the split baud issue, but without sources for whatever can be dropped in..  it was there, and probably to deal with modem backchannels, was that all though?

I'm reading...

**Reply by KC5VDJ** [reply]
I'm also looking at how this fits into a more general thing I have going on for a comprehensive terminal emulator, that will at least support tcp/ip and serial, adding Bluetooth and other channels as well later.  This is why I've been collecting terminal manuals.  As such, yes, how the oddballs work is important to me, and this makes a vt-180 look like minimalist.

**Comment by KC5VDJ** [comment]
Good memory.....  12.7 mm squares for the touch.

**Comment by KC5VDJ** [comment]
Also, even hexdumps (commented assembly?) of the ROMs of all models that can be achieved on, preferably four sets from each model, use a voting system, at least with proms (prom/eprom tech seems to not be aging well).  Mask rom is mask rom, short of a catastrophic failure, one would do, maybe a compared trio (either trust it or don't, mask'll prob be good, bitrot is real with eproms tho).

**Comment by KC5VDJ** [comment]
ASCII loader....  24 bit absolute address for load, but only lower 16 used (avoiding 42k of even that).  Was the high byte used for anything?  Seems so wasteful, someone had to have used it for something.

**Reply by KC5VDJ** [reply]
Assuming the controlware talks the same terminal commands, could it be used for scrolling regions, edit regions, possibly with scrolling, possibly text-based overlays/windows, and validation?  All of that could easily fit in 22k.  The original thin client?
